src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs

```diff
@@ -187,13 +254,17 @@ private void ReportPendingWorkItemCount()
                 private async Task RunAnalyzersAsync<T>(
                     ImmutableArray<IIncrementalAnalyzer> analyzers,
                     T value,
+                    WorkItem? workItem,
                     Func<IIncrementalAnalyzer, T, CancellationToken, Task> runnerAsync,
                     CancellationToken cancellationToken)
                 {
                     using var evaluating = _registration.ProgressReporter.GetEvaluatingScope();
 
                     ReportPendingWorkItemCount();
-                    foreach (var analyzer in analyzers)
+
+                    // Check if the work item is specific to some incremental analyzer(s).
+                    var analyzersToExecute = workItem?.GetApplicableAnalyzers(analyzers) ?? analyzers;
```

Comment:
i'm a bit confused for when workItem is null... if workItem is null... why run anything?  the relationship between these concepts here isn't clear to me.

Reply:
Yeah, I will change the API to pass in applicable analyzers for the single case where work item is null at the call site.

Reply:
I just switched the single call site which was passing null to use a dummy work item, and things work just fine by switching workItem to be a non-nullable parameter.

Reply:
Love it!